staging/src/k8s.io/apimachinery/pkg/runtime/schema/group_version_test.go

```diff
@@ -134,3 +135,47 @@ func TestKindForGroupVersionKinds(t *testing.T) {
 		}
 	}
 }
+
+func TestParseKindArg(t *testing.T) {
+	tests := []struct {
+		input string
+		gvk   *GroupVersionKind
+		gk    GroupKind
+	}{
+		{input: "Pod", gk: GroupKind{Kind: "Pod"}},
+		{input: ".apps", gk: GroupKind{Group: "apps"}},
+		{input: "Pod.", gk: GroupKind{Kind: "Pod"}},
+		{input: "Pod.apps", gk: GroupKind{Group: "apps", Kind: "Pod"}},
+		{input: "Pod.v1.apps", gvk: &GroupVersionKind{Group: "apps", Version: "v1", Kind: "Pod"}, gk: GroupKind{Group: "v1.apps", Kind: "Pod"}},
+	}
+	for i, test := range tests {
+		t.Run(fmt.Sprintf("%s", test.input), func(t *testing.T) {
```

Comment:
You should be able to just do `test.Run(test.input, func(***`, right? test.input is already a string.